### PR TITLE
Synnefo/Pithos+ object store backend

### PR DESCRIPTION
**Pull request by saxtouri**
An implementation of the storage backend class (lib/galaxy/objectstore: ObjectStore)  for Synnefo/Pithos+ object store ( https://www.synnefo.org/docs/synnefo/latest/pithos.html ) for Galaxy.

The implementation is using "kamaki", which is the Synnefo API client ( https://www.synnefo.org/docs/kamaki/latest/ ).

### CODE REVIEW

**Comment by saxtouri** [comment]
The tests do not fail on my system. It seems that the tests fail on travis on the "import kamaki" line, but I have added "kamaki" to "conditional-requirements.txt" and this seems to suffice on my tests.

Any suggestions on what went wrong?

**Comment by bgruening** [comment]
@saxtouri I guess Galaxy needs a wheel to make this work. Can you create one with starforge? https://github.com/galaxyproject/starforge/blob/master/wheels/build/wheels.yml

**Comment by martenson** [comment]
The wheels for Kamaki are now available: https://depot.galaxyproject.org/starforge/wheels/build-165/

File: lib/galaxy/dependencies/conditional-requirements.txt
Comment (martenson) [inline lib/galaxy/dependencies/conditional-requirements.txt:17]: this also needs something like this:
```
def check_kamaki( self ):
        return 'pithos' in self.object_stores
```

around [here](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/dependencies/__init__.py#L108)

**Review by martenson — CHANGES_REQUESTED** [review]
please see above

File: lib/galaxy/dependencies/conditional-requirements.txt
Comment (saxtouri) [reply lib/galaxy/dependencies/conditional-requirements.txt:17]: Done! Thanks!

**Comment by martenson** [comment]
The failing test is due to our (lack of) architecture for testing conditional dependencies. I am trying to work it out.

**Comment by martenson** [comment]
@galaxybot test this

**Comment by saxtouri** [comment]
@galaxybot test this

**Comment by nsoranzo** [comment]
@galaxybot test this

**Review by martenson — APPROVED** [review]

**Comment by martenson** [comment]
From my standpoint LGTM. However someone more fluent in object store should have a look here before this gets merged.

**Comment by bgruening** [comment]
@dannon @natefoo would be great to get your input here :)
@saxtouri thanks, great work!

**Comment by dannon** [comment]
@saxtouri This looks great to me, but prior to incorporation, would you mind dropping the copyright statement?  It'd be fine to change it to an attribution statement/comment if you'd like (so, something like "This work originally developed by ```<people>``` at ```<institution>```"), and add yourself to https://github.com/galaxyproject/galaxy/blob/dev/CONTRIBUTORS.md, but code contributions should fall under the general project copyright and licensing terms.

(and, apologies for not thinking about that bit until I was about to merge!)

**Comment by jmchilton** [comment]
Ping @saxtouri again - is that copyright request reasonable?

**Comment by saxtouri** [comment]
Sorry for the delay, of course it is reasonable, I'm updating this today.

**Comment by saxtouri** [comment]
OK, I removed the copyright statement. I apologize for the delay. GRNET didn't really have a problem with this, the statement was put there by scripts I use.

**Comment by dannon** [comment]
Thank you @saxtouri!  I'll run some final tests and merge this in a second.

**Comment by martenson** [comment]
Thanks @saxtouri , very nice contribution!

**Pull request merged**
